include/solvers/MFEMHypreBoomerAMG.h

```diff
@@ -0,0 +1,24 @@
+#pragma once
+#include "MFEMSolverBase.h"
+#include "mfem.hpp"
+#include <memory>
+
+/**
+ * Wrapper for mfem::HypreGMRES solver.
+ */
+class MFEMHypreBoomerAMG : public MFEMSolverBase
+{
+public:
+  static InputParameters validParams();
+
+  MFEMHypreBoomerAMG(const InputParameters &);
+
+  /// Returns a shared pointer to the instance of the Solver derived-class.
+  std::shared_ptr<mfem::Solver> getSolver() const override { return _preconditioner; }
+
+protected:
+  void constructSolver(const InputParameters & parameters) override;
+
+private:
+  std::shared_ptr<mfem::HypreBoomerAMG> _preconditioner{nullptr};
```

Comment:
At least according to the [BoomerAMG documentation](https://hypre.readthedocs.io/en/latest/solvers-boomeramg.html), it can be used as a solver or a preconditioner. Is there any particular reason we're only allowing it to be a preconditioner here?

Reply:
Every solver we have can be used in the preconditioner block (so it should work regardless, as they're just mfem::Solvers anyway) - but I have renamed the private variables used in MFEMHypreBoomerAMG to match the others. I've also added the minimal set of solver options required to use it as a standalone solver when needed.